graphdatascience/query_runner/neo4j_query_runner.py

```diff
@@ -14,28 +14,84 @@
 from ..error.endpoint_suggester import generate_suggestive_error_message
 from ..error.unable_to_connect import UnableToConnectError
 from ..server_version.server_version import ServerVersion
+from ..version import __version__
 from .cypher_graph_constructor import CypherGraphConstructor
 from .graph_constructor import GraphConstructor
 from .query_runner import QueryRunner
+from graphdatascience.error.gds_not_installed import GdsNotFound
 
 
 class Neo4jQueryRunner(QueryRunner):
+    _AURA_DS_PROTOCOL = "neo4j+s"
     _LOG_POLLING_INTERVAL = 0.5
     _NEO4J_DRIVER_VERSION = ServerVersion.from_string(neo4j.__version__)
 
+    @staticmethod
+    def create(
+        endpoint: Union[str, neo4j.Driver, QueryRunner],
+        auth: Optional[Tuple[str, str]] = None,
+        aura_ds: bool = False,
+        database: Optional[str] = None,
+        bookmarks: Optional[Any] = None,
+        arrow: bool = True,
+    ) -> "QueryRunner":
+        query_runner: "QueryRunner"
+        if isinstance(endpoint, str):
+            config: Dict[str, Any] = {"user_agent": f"neo4j-graphdatascience-v{__version__}"}
+
+            if aura_ds:
+                Neo4jQueryRunner._configure_aura(endpoint, config)
+
+            driver = neo4j.GraphDatabase.driver(endpoint, auth=auth, **config)
+
+            query_runner = Neo4jQueryRunner(driver, auto_close=True, bookmarks=bookmarks, config=config)
+
+        elif isinstance(endpoint, QueryRunner):
+            if arrow:
+                raise ValueError("Arrow cannot be used if the QueryRunner is provided directly")
+
+            query_runner = endpoint
+
+        else:
+            driver = endpoint
+            query_runner = Neo4jQueryRunner(driver, auto_close=False, bookmarks=bookmarks)
```

Comment:
I guess we could do `query_runner = if ... elif ... else` to avoid the weird `query_runner: "QueryRunner" at the top?

Reply:
nope, not in Python

Reply:
using the `from __future__ import annotations` gets rid of the weird "QueryRunner" :)